torch/_inductor/mkldnn_lowerings.py

```diff
@@ -544,24 +547,75 @@ def qlinear_unary(
             bias_dtype = None if bias is None else bias.get_dtype()
 
             choices: List[ChoiceCaller] = []
+
             if use_max_autotune():
                 *_, layout, x, packed_weight = mm_args(
                     x, packed_weight, layout=layout, out_dtype=output_dtype
                 )
+
+                def _is_sym_quantized_wgt():
+                    # weight zero point may not be a ConstantBuffer, but a computeBuffer instead.
+                    # but it is empty/dummy, which means the weight is symmetrically quantized.
+                    # We've already established by now that packed weights are int8, so if w_zp is not dummy,
+                    # then all its values must be zeros if weights are int8 symmetrically quantized.
+                    nonlocal w_zp
+                    if (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(w_zp),
+                            ir.ComputedBuffer,
+                        )
+                        and w_zp.get_numel() == 0
+                    ) or (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(w_zp),
+                            ir.ConstantBuffer,
+                        )
+                        and torch.equal(
+                            torch.zeros_like(V.graph.constants[w_zp.get_name()]),
+                            V.graph.constants[w_zp.get_name()],
+                        )
+                    ):
+                        return True
+                    else:
+                        return False
+
+                def _is_act_comptaible_with_onednn():
+                    # oneDNN doesn't currently have an optimized kernel for a vector activation zero-point.
+                    # So, if the activation zero point is a vector, we won't support it with oneDNN.
+                    # We can support the provided activation if:
+                    # 1. if x_zp is dummy (empty tensor) created to match an Inductor pattern
+                    # 2. If the activation is per-tensor quantized (x may even be uint8 in that case).
```

Comment:
Add the scenario which belong to each case?

Reply:
Sorry, which scenarios are you referring to? I think they've already been described.